applications/services/input/input.c

```diff
@@ -75,95 +72,187 @@ const char* input_get_type_name(InputType type) {
     case InputTypeRepeat:
         return "Repeat";
     default:
-        return "Unknown";
+        furi_crash();
     }
 }
 
-int32_t input_srv(void* p) {
-    UNUSED(p);
-
-    const FuriThreadId thread_id = furi_thread_get_current_id();
-    FuriPubSub* event_pubsub = furi_pubsub_alloc();
-    uint32_t counter = 1;
-    furi_record_create(RECORD_INPUT_EVENTS, event_pubsub);
+static void input_isr_key(void* context) {
+    InputSrv* instance = context;
+    furi_semaphore_release(instance->input_semaphore);
```

Comment:
Interrupts may be missed in some edge cases

Reply:
fix